src/node/mini_miner.cpp

```diff
@@ -0,0 +1,368 @@
+    //               │                 │                           P1_ASFR:  9.84 s/vB
+    //               └──────▲───▲──────┘                           P2_ASFR:  2.47 s/vB
+    //                      │   │                                   C_ASFR: 10.27 s/vB
+    // ┌───────────────┐    │   │    ┌──────────────┐
+    // │               ├────┘   └────┤              │             ⇒ C_FR < TFR < C_ASFR
+    // │   Parent 1    │             │   Parent 2   │
+    // │    200 vB     │             │    200 vB    │
+    // │  17000 sats   │             │   3000 sats  │
+    // │     85 s/vB   │             │     15 s/vB  │
+    // │               │             │              │
+    // └───────────▲───┘             └───▲──────────┘
+    //             │                     │
+    //             │    ┌───────────┐    │
+    //             └────┤           ├────┘
+    //                  │   Child   │
+    //                  │  100 vB   │
+    //                  │  900 sats │
+    //                  │    9 s/vB │
+    //                  │           │
+    //                  └───────────┘
+    //
+    // We therefore calculate both the bump fee that is necessary to elevate
+    // the individual transaction to the target feerate:
+    //         target_feerate × tx_size - tx_fees
+    // and the bump fee that is necessary to bump the entire ancestor set to
+    // the target feerate:
+    //         target_feerate × ancestor_set_size - ancestor_set_fees
+    // By picking the maximum from the two, we ensure that a transaction meets
+    // both criteria.
+    for (const auto& [txid, outpoints] : m_requested_outpoints_by_txid) {
+        auto it = m_entries_by_txid.find(txid);
+        Assume(it != m_entries_by_txid.end());
+        if (it != m_entries_by_txid.end()) {
+            Assume(target_feerate.GetFee(it->second.GetSizeWithAncestors()) > std::min(it->second.GetModifiedFee(), it->second.GetModFeesWithAncestors()));
+            CAmount bump_fee_with_ancestors = target_feerate.GetFee(it->second.GetSizeWithAncestors()) - it->second.GetModFeesWithAncestors();
+            CAmount bump_fee_individual = target_feerate.GetFee(it->second.GetTxSize()) - it->second.GetModifiedFee();
+            const CAmount bump_fee{std::max(bump_fee_with_ancestors, bump_fee_individual)};
+            Assume(bump_fee >= 0);
+            for (const auto& outpoint : outpoints) {
+                m_bump_fees.emplace(outpoint, bump_fee);
+            }
+        }
+    }
+    return m_bump_fees;
+}
+
+std::optional<CAmount> MiniMiner::CalculateTotalBumpFees(const CFeeRate& target_feerate)
+{
+    if (!m_ready_to_calculate) return std::nullopt;
+    // Build a block template until the target feerate is hit.
+    BuildMockTemplate(target_feerate);
+
+    // All remaining ancestors that are not part of m_in_block must be bumped, but no other relatives
+    std::set<MockEntryMap::iterator, IteratorComparator> ancestors;
+    {
```

Comment:
nit: Seems like this pair of curly-braces and the extra-indent is a left-over from earlier code (probably involving a `LOCK`?) and can just be removed.